src/NHibernate/Engine/ISessionImplementor.cs

```diff
@@ -225,7 +225,17 @@ public partial interface ISessionImplementor
 		/// <summary> Retrieves the configured event listeners from this event source. </summary>
 		EventListeners Listeners { get; }
 
-		int DontFlushFromFind { get; }
+		/// <summary>
+		/// Is auto-flush suspended?
+		/// </summary>
+		bool DontFlushFromFind { get; }
+
+		/// <summary>
+		/// Suspend auto-flushing, yielding a disposable to dispose when auto flush should be restored. Supports
+		/// being called multiple times.
+		/// </summary>
+		/// <returns>A disposable to dispose when auto flush should be restored.</returns>
+		IDisposable SuspendAutoFlush();
```

Comment:
Needing it from an other internal class than the session, and factoring it by the way.